clients/identity/authentication.go

```diff
@@ -26,19 +26,22 @@ type Client common.APIClient
 
 // Clienter provides an interface to checking identity of incoming request
 type Clienter interface {
-	CheckRequest(req *http.Request) (context.Context, int, error)
+	CheckRequest(req *http.Request) (context.Context, int, authError, httpError)
```

Comment:
hate to be pedantic (...), but can you reconsider these errors for clarity, please: how about
`..., authFailure, error)`
i.e.:
- keep `error` for *general* http/json errors (serious, general, external errors)
- use `authFailure` (type `error`) to indicate whether auth failed

the crucial problem, for me, being that your `httpError` can also results from a json marshall error (which _is_ http related, but seems more general to me), but using just `error` emphasises a systemic failure rather than a semantic (authFailure) issue.  Idiomatic?